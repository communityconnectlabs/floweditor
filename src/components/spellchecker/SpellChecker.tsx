import React from 'react';
import CheckboxElement from '../form/checkbox/CheckboxElement';
import HelpIcon from '../helpicon/HelpIcon';
import styles from './SpellChecker.module.scss';
import variables from 'variables.module.scss';

export interface SpellCheckerProps {
  enabledSpell: boolean;
  onEnabledChange(): void;
  spellSensitivity: string;
  onSensitivityChange(event: React.FormEvent<HTMLInputElement>): void;
}

export class SpellChecker extends React.Component<SpellCheckerProps> {
  public render(): JSX.Element {
    return (
      <div className={styles.spellcheckerWrap}>
        <div>
          <CheckboxElement
            name="Enable Spell Checker"
            checked={this.props.enabledSpell}
            description="Enable spelling correction"
            onChange={this.props.onEnabledChange}
          />
          <HelpIcon
            iconColor={variables.dark_blue}
            iconSize="15px"
            dataFor="enableSpell"
            bigTooltip
          >
            <p>
              This feature will automatically correct typos and misspellings by users in multiple
              languages. The language for the spell checker is based on the language set for the
              user. Corrections occur before the categorization rules. For example, if user types
              “thankks” it would be corrected to “thanks.” The word “thanks” would then be used for
              the categorization. The corrected text is available as a flow variable with
              @flow.variable.corrected
            </p>
            <p>
              We encourage you to test misspellings in order to tune the sensitivity for your use
              case. For more information and a list of languages, see our{' '}
              <a
                rel="noopener noreferrer"
                target="_blank"
                href="https://docs.microsoft.com/en-us/azure/cognitive-services/bing-spell-check/language-support"
              >
                help page
              </a>
              .
            </p>
          </HelpIcon>
        </div>
        {this.props.enabledSpell && (
          <div className={styles.rangerContainer}>
            <div>
              <span>Correction Sensitivity</span>
              <HelpIcon
                iconColor={variables.dark_blue}
                iconSize="15px"
                dataFor="spellSensitivity"
                bigTooltip
              >
                <p>
                  This slider bar enables you to control the sensitivity of the correction. Like all
                  spell checkers, it does not always make the correction you would want it to make.
                  The higher the sensitivity, the more corrections will be made. Lower sensitivity
                  will only make more obvious corrections.
                </p>
                <p>
                  We encourage you to test misspellings in order to tune the sensitivity for your
                  usecase. For more information and a list of languages, see our{' '}
                  <a
                    rel="noopener noreferrer"
                    target="_blank"
                    href="https://docs.microsoft.com/en-us/azure/cognitive-services/bing-spell-check/language-support"
                  >
                    help page
                  </a>
                  .
                </p>
              </HelpIcon>
            </div>
            <div className={styles.inputWrap}>
              <div>Low</div>
              <input
                type="range"
                min="70"
                max="99"
                value={this.props.spellSensitivity}
                onChange={this.props.onSensitivityChange}
                className={styles.rangerInput}
              />
              <div>High</div>
            </div>
            <div className={styles.sensitivity}>
              {`${this.props.spellSensitivity}% confidence in corrections`}
            </div>
          </div>
        )}
      </div>
    );
  }
}
