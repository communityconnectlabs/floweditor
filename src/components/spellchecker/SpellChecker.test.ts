import { composeComponentTestUtils } from 'testUtils';
import { SpellChecker, SpellCheckerProps } from './SpellChecker';

const baseProps: SpellCheckerProps = {
  enabledSpell: true,
  onEnabledChange: jest.fn(),
  spellSensitivity: '80',
  onSensitivityChange: jest.fn()
};

const { setup } = composeComponentTestUtils(SpellChecker, baseProps);

describe('SpellChecker component', () => {
  it('should render checkbox and ranger', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render only checkbox', () => {
    const { wrapper } = setup(false, { $merge: { enabledSpell: false } });
    expect(wrapper).toMatchSnapshot();
  });
});
