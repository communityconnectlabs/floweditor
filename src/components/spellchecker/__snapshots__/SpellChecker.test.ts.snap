// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpellChecker component should render checkbox and ranger 1`] = `
<div
  className="spellcheckerWrap"
>
  <div>
    <CheckboxElement
      checked={true}
      description="Enable spelling correction"
      name="Enable Spell Checker"
      onChange={[MockFunction]}
    />
    <HelpIcon
      bigTooltip={true}
      dataFor="enableSpell"
      iconColor="dark_blue"
      iconSize="15px"
    >
      <p>
        This feature will automatically correct typos and misspellings by users in multiple languages. The language for the spell checker is based on the language set for the user. Corrections occur before the categorization rules. For example, if user types “thankks” it would be corrected to “thanks.” The word “thanks” would then be used for the categorization. The corrected text is available as a flow variable with @flow.variable.corrected
      </p>
      <p>
        We encourage you to test misspellings in order to tune the sensitivity for your use case. For more information and a list of languages, see our
         
        <a
          href="https://docs.microsoft.com/en-us/azure/cognitive-services/bing-spell-check/language-support"
          rel="noopener noreferrer"
          target="_blank"
        >
          help page
        </a>
        .
      </p>
    </HelpIcon>
  </div>
  <div
    className="rangerContainer"
  >
    <div>
      <span>
        Correction Sensitivity
      </span>
      <HelpIcon
        bigTooltip={true}
        dataFor="spellSensitivity"
        iconColor="dark_blue"
        iconSize="15px"
      >
        <p>
          This slider bar enables you to control the sensitivity of the correction. Like all spell checkers, it does not always make the correction you would want it to make. The higher the sensitivity, the more corrections will be made. Lower sensitivity will only make more obvious corrections.
        </p>
        <p>
          We encourage you to test misspellings in order to tune the sensitivity for your use case. For more information and a list of languages, see our
           
          <a
            href="https://docs.microsoft.com/en-us/azure/cognitive-services/bing-spell-check/language-support"
            rel="noopener noreferrer"
            target="_blank"
          >
            help page
          </a>
          .
        </p>
      </HelpIcon>
    </div>
    <div
      className="inputWrap"
    >
      <div>
        High
      </div>
      <input
        className="rangerInput"
        max="99"
        min="70"
        onChange={[MockFunction]}
        type="range"
        value="80"
      />
      <div>
        Low
      </div>
    </div>
    <div
      className="sensitivity"
    >
      80% confidence in corrections
    </div>
  </div>
</div>
`;

exports[`SpellChecker component should render only checkbox 1`] = `
<SpellChecker
  enabledSpell={false}
  onEnabledChange={[MockFunction]}
  onSensitivityChange={[MockFunction]}
  spellSensitivity="80"
>
  <div
    className="spellcheckerWrap"
  >
    <div>
      <CheckboxElement
        checked={false}
        description="Enable spelling correction"
        name="Enable Spell Checker"
        onChange={[MockFunction]}
      >
        <label
          className="label"
          onClick={[Function]}
        >
          <span
            className="fe-square"
            data-spec="checkbox"
          />
          <div
            className="description_solo"
            data-spec="description"
          >
            Enable spelling correction
          </div>
        </label>
      </CheckboxElement>
      <HelpIcon
        bigTooltip={true}
        dataFor="enableSpell"
        iconColor="dark_blue"
        iconSize="15px"
      >
        <div
          className="icon"
          data-for="enableSpell"
          data-tip=""
          style={
            Object {
              "color": "dark_blue",
              "fontSize": "15px",
            }
          }
        >
          <span
            className="fe-help"
          />
        </div>
        <ReactTooltip
          className="tooltip bigTooltip"
          clickable={false}
          delayHide={300}
          effect="solid"
          id="enableSpell"
          insecure={true}
          multiline={true}
          resizeHide={true}
          wrapper="div"
        >
          <div
            className="__react_component_tooltip place-top type-dark"
            data-id="tooltip"
            id="enableSpell"
          >
            <p>
              This feature will automatically correct typos and misspellings by users in multiple languages. The language for the spell checker is based on the language set for the user. Corrections occur before the categorization rules. For example, if user types “thankks” it would be corrected to “thanks.” The word “thanks” would then be used for the categorization. The corrected text is available as a flow variable with @flow.variable.corrected
            </p>
            <p>
              We encourage you to test misspellings in order to tune the sensitivity for your use case. For more information and a list of languages, see our
               
              <a
                href="https://docs.microsoft.com/en-us/azure/cognitive-services/bing-spell-check/language-support"
                rel="noopener noreferrer"
                target="_blank"
              >
                help page
              </a>
              .
            </p>
          </div>
        </ReactTooltip>
      </HelpIcon>
    </div>
  </div>
</SpellChecker>
`;
